mk/spksrc.wheel-compile.mk

```diff
@@ -0,0 +1,148 @@
+### Wheel rules
+# Compile wheels for modules listed in WHEELS. 
+#
+# Targets are executed in the following order:
+#  wheel_compile_msg_target
+#  pre_wheel_compile_target   (override with PRE_WHEEL_COMPILE_TARGET)
+#  wheel_compile_target       (override with WHEEL_COMPILE_TARGET)
+#  post_wheel_compile_target  (override with POST_WHEEL_COMPILE_TARGET)
+# Variables:
+#  REQUIREMENT             Requirement formatted wheel information
+#  WHEEL_NAME              Name of wheel to process
```

Comment:
how to document, that WHEEL_NAME must not contain `-`?
Since name and version are separated by `-` the name must not contain `-`.

can we add a check here 
or fail if name has `-` instead of `_` 
or implement an auto renaming?

or can we still use `=` as version separator? (can't remeber which make file is affected).

Reply:
Unless I don't understand your comment: 
`WHEEL_NAME` really is just the wheel name.
`WHEEL_VERSION` really is just the version
`REQUIREMENT` is `WHEEL_NAME==WHEEL_VERSION` most of the time, although sometime it can be a `git+https://` type url with a `#egg=WHEEL_NAME==WHEL_VERSION`.

Although it did made me think of something else... said parameter is "WHEELS" similarly to variable in usage in our makefile, which is fine.  But for the crossenv I'm using "WHEEL" with an expectation that it takes a `<name>-<version>` - version being optional in this case. 

So, potentially:
1. allow using `-` in WHEELS?
2. make crossenv use the same variable name? or something completely different because `WHEEL` and `WHEELS` could be confusing^

Reply:
OK, found the related code in `spksrc.crossenv.mk`
lines 62 and 63:
```
else ifneq ($(wildcard $(CROSSENV_CONFIG_PATH)/requirements-$(firstword $(subst -, ,$(WHEEL))).txt),)
CROSSENV_BUILD_WHEEL = $(firstword $(subst -, ,$(WHEEL)))
```

The expression ` $(firstword $(subst -, ,$(WHEEL)))` does not match the wheel name, when it contains `-`, but it works when `_` is used instead.

we could use `=` or `==` to separate name and version in `WHEEL`-variables or we ensure that we split name and version at the most right `- `.
But even we do this, the wheel specific cross-env definition file name must match the wheel name in the requirements files of the packages.

Reply:
I suggest the following to simplify things: with this new code `spksrc.wheel.mk` will always manage `WHEEL_NAME` and `WHEEL_VERSION` variables.  I suggest replacing the confusing `WHEEL` only used with the newly introduces dynamic `crossenv` code for using those two variables instead.

I'll update the code for this.  Also this way both wheel and crossenv will be aligned.

Reply:
code now adapted for this.